### PR TITLE
Dev/178: [POST] /api/exams/{examId}/groups

### PR DESCRIPTION
**Pull request by Johnny850807**
Resolve: #178

### CODE REVIEW

File: Domain/Primitives/src/main/java/tw/waterball/judgegirl/primitives/exam/Exam.java [hunk 1]
```diff
@@ -84,4 +86,12 @@ public void updateQuestion(Question question) {
         updatedQuestion.setQuota(question.getQuota());
         updatedQuestion.setScore(question.getScore());
     }
+
+    public void addStudentsAsExaminees(List<Student> students) {
+        addExaminees(mapToList(students, student -> new Examinee(id, student.getId())));
+    }
+
+    public void addExaminees(List<Examinee> examinees) {
```
Comment (Wally5077) [inline Domain/Primitives/src/main/java/tw/waterball/judgegirl/primitives/exam/Exam.java:94]:
```Java
public void addExaminees(Collection<Examinee> examinees)
```

File: Spring-Boot/Spring-Boot-Academy/src/test/java/tw/waterball/judgegirl/springboot/academy/controllers/ExamControllerTest.java [hunk 2]
```diff
@@ -126,12 +121,15 @@ private void fakeProblemServiceDriver() {
...
-        student1.setId(STUDENT_A_ID);
-        Student student2 = new Student("studentB", STUDENT_B_EMAIL, "passwordB");
-        student2.setId(STUDENT_B_ID);
-        studentServiceDriver.addStudent(student1);
-        studentServiceDriver.addStudent(student2);
+        Student studentA = new Student("studentA", STUDENT_A_EMAIL, "passwordA");
+        studentA.setId(STUDENT_A_ID);
+        Student studentB = new Student("studentB", STUDENT_B_EMAIL, "passwordB");
+        studentB.setId(STUDENT_B_ID);
+        Student studentC = new Student("studentC", STUDENT_C_EMAIL, "passwordB");
```
Comment (Wally5077) [inline Spring-Boot/Spring-Boot-Academy/src/test/java/tw/waterball/judgegirl/springboot/academy/controllers/ExamControllerTest.java:128]: passwordB -> passwordC

**Review by Wally5077 — CHANGES_REQUESTED** [review]
Review done

**Review by Wally5077 — APPROVED** [review]
LGTM

File: Spring-Boot/Spring-Boot-Academy/src/test/java/tw/waterball/judgegirl/springboot/academy/controllers/ExamControllerTest.java [hunk 3]
```diff
@@ -198,32 +196,50 @@ void WhenUpdateExamWithNonExistingExamId_ShouldRespondNotFound() throws Exceptio
...
 
         List<String> errorEmails = getBody(
-                createExaminees(exam.id, STUDENT_A_EMAIL, STUDENT_B_EMAIL, "studentC@example.com")
+                addExaminees(exam.id, STUDENT_A_EMAIL, STUDENT_B_EMAIL, "studentZ@example.com")
                         .andExpect(status().isOk()), new TypeReference<>() {
                 });
 
         assertEquals(1, errorEmails.size());
-        assertEquals("studentC@example.com", errorEmails.get(0));
+        assertEquals("studentZ@example.com", errorEmails.get(0));
```
Comment (ShawnSWu) [inline Spring-Boot/Spring-Boot-Academy/src/test/java/tw/waterball/judgegirl/springboot/academy/controllers/ExamControllerTest.java:227]: Why not declare local variable String studentCEmail?

**Review by ShawnSWu — CHANGES_REQUESTED** [review]

**Review by timt1028 — APPROVED** [review]
LGTM

File: Spring-Boot/Spring-Boot-Academy/src/test/java/tw/waterball/judgegirl/springboot/academy/controllers/ExamControllerTest.java (hunk 3 above)
Comment (Johnny850807) [reply Spring-Boot/Spring-Boot-Academy/src/test/java/tw/waterball/judgegirl/springboot/academy/controllers/ExamControllerTest.java:227]: unnecessary

**Review by ShawnSWu — APPROVED** [review]

**Pull request merged**